pbj-core/pbj-runtime/src/test/java/com/hedera/pbj/runtime/ProtoWriterToolsTest.java

```diff
@@ -986,6 +1059,268 @@ void testSizeOfDelimited() {
         assertEquals(MIN_LENGTH_VAR_SIZE + TAG_SIZE + length, sizeOfDelimited(definition, length));
     }
 
+    private static Stream<Arguments> provideWriteIntegerListArguments() {
+        return Stream.of(
+                Arguments.of(INT32, 29, false),
+                Arguments.of(UINT32, 24, false),
+                Arguments.of(SINT32, 21, true),
+                Arguments.of(FIXED32, 32, false),
+                Arguments.of(SFIXED32, 32, false)
+        );
+    }
+
+    @ParameterizedTest
+    @MethodSource("provideWriteIntegerListArguments")
+    void testWriteIntegerList(final FieldType type, final int expectedSize, final boolean zigZag) {
+        final FieldDefinition definition = createRepeatedFieldDefinition(type);
+
+        final long start = bufferedData.position();
+        ProtoWriterTools.writeIntegerList(bufferedData, definition, List.of(
+                0x0f,
+                0xff,
+                0xfff,
+                0xffff,
+                0xfffff,
+                0xffffff,
+                0xfffffff,
+                0xffffffff
+        ));
+        final long finish = bufferedData.position();
+
+        int tag = bufferedData.getVarInt(start, false);
+        assertEquals(WIRE_TYPE_DELIMITED.ordinal(), tag & 0x3);
```

Comment:
`ProtoParserTools` has a constant for this 0x03 value, it can be used here. Also, the constant seems to be useful not only while parsing, but also on writing, so it can be moved to `ProtoConstants`, maybe?

Reply:
That's a very good catch. I think it should actually be 0x7 because there's 3 bits in the wire type. I'll use a common constant for that.